wandb/weave
# chore(artifacts): update artifact fs - add a new uri by id class to allow querying artifact manifest via id

Use case: 
- Upon **linking** an artifact from a team's project to a collection in a global registry (essentially org-level projects), we'd like to make it accessible it to all users of the registry. However, the current method of querying the artifact files via Weave1 makes that impossible since we're dependent on the entity name of the source artifact, which is hidden from users that are not in that team.
- This PR is an attempt to change the artifact_version_file_op to allow **querying for the artifact and its files only using its ID**, so that users across the org can view and use the artifact while keeping the source artifact's project and team (entity) names hidden from users that don't belong to the same team.

Corresponding W0 change: https://github.com/wandb/core/pull/20157

* Added some tests in `test_wb.py`

**Test workflow:**

If you don't want to go through the listed steps, add yourself to the `dev-team-ibindlish-org` and try navigating to 
`https://app.wandb.test/registry/dataset?selectionPath=dev-team-ibindlish-org%2Fwandb-registry-dataset%2Ftesting-show-links&view=membership&tab=files&version=v0`. You should be able to see the list of files in the artifact. (Make sure you update the Weave_wandb_cookie in your dev env so that its not using an admin cookie)

1. Login to an org with at least 2 teams - team1 and team2 and 2 users - user1 and user2, where user1 is a member of team1 and user2 is a member of team2. Login as user1
2. Enable the global registry feature flag for this org
3. Log an artifact in team1 and link it to a collection in one of the global registries at the org-level
4. Login as user2 (update `WEAVE_WANDB_COOKIE` in .admin.env) and navigate directly to the files tab of the linked artifact. (The versions page of this registry collection will crash right now since user2 isn't allowed to read the run info which is displayed in the `PanelArtifactHistory` (aka Versions) table
5. As user2, you should be able to see the files of the linked artifact, even though user2 is not a member of team1 by navigating directly to the `Files` tab of the linked version. Navigating through some of the other artifact tabs will throw an error since on the fe we query for run metrics which are hidden to unauthorized users. You can go to:

`https://app.wandb.test/registry/<registryName>selectionPath=<org-entity-name>%2Fwandb-registry-<registryName>%2F<collectionName>&view=membership&tab=files&version=v0`

Note: org-entity-name is almost always the same as the org name.

Previously you'd would only see an empty file browser. 

**Also verified artifact auth**
Using Postman, I tried to query the artifact, its manifest and its files, via just its ID but with a Cookie that belonged to a user outside the artifact's organization. The response was `null` since the artifact auth failed.
